### PR TITLE
Add Version endpoint that uses internal git properties version. Depricated old endpoint

### PR DESCRIPTION
**Pull request by cfkoehler**
This new endpoint does not need to spin up and emissary server to get version details.
To view the output you can curl a running emissary server:
`curl -u emissary:emissary123 --anyauth http://localhost:8001/api/versionDetails | jq`
which responds with:
```
{
  "errors": [],
  "response": {
    "Build Date": "2022-10-14T22:42:52+0000",
    "Build Host": "ip-10-113-15-113.evoforge.org",
    "Git Abbrev Hash": "d308cf5",
    "Version": "7.10.0-SNAPSHOT"
  }
}
```

### CODE REVIEW

**Comment by cfkoehler** [comment]
I am still working on unit test but am having problems trying to mock GitRepositoryState.class so might need help with that.

File: src/main/java/emissary/server/api/Version.java [hunk 1]
```diff
@@ -81,4 +87,36 @@ private MapResponseEntity lookupVersion() {
...
     }
 
+    @GET
+    @Path("/versionSimple")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response versionSimple() {
+        MapResponseEntity entity = new MapResponseEntity();
+        try {
+            GitRepositoryState gitRepoState = GitRepositoryState.getRepositoryState();
+            entity.addKeyValue("Version", gitRepoState.getBuildVersion());
```
Comment (cfkoehler) [inline src/main/java/emissary/server/api/Version.java:97]: The new response does not return the host:port of the server. I figured if the endpoint is being hit then then the user already knows the server name and port and it does not need to be returned. Also not sure how to get those values :)

File: src/test/java/emissary/server/api/EmissaryApiTest.java [hunk 2]
```diff
@@ -178,6 +178,25 @@ void version() {
         }
     }
 
+     @Test
+     void versionSimple() {
+     GitRepositoryState mockRepoState = mock(GitRepositoryState.class);
```
Comment (pechrista) [inline src/test/java/emissary/server/api/EmissaryApiTest.java:183]: getting build failures with not being able to find GitRepositoryState... need to make constructor public in GitRepositoryState.java

File: src/test/java/emissary/server/api/EmissaryApiTest.java [hunk 3]
```diff
@@ -178,6 +178,25 @@ void version() {
...
+     doReturn("1.0.0").when(mockRepoState).getBuildVersion();
+     try (Response response = target("versionSimple").request().get()) {
+     assertEquals(200, response.getStatus());
+     MapResponseEntity entity = response.readEntity(MapResponseEntity.class);
+     assertTrue(entity.getErrors().isEmpty());
+     assertFalse(entity.getResponse().isEmpty());
+     assertTrue(entity.getResponse().containsKey("Version"));
+     assertEquals("1.0.0", entity.getResponse().get("Version"));
```
Comment (pechrista) [inline src/test/java/emissary/server/api/EmissaryApiTest.java:191]: getting test version as its expecting the emissary version

File: src/main/java/emissary/server/api/Version.java (hunk 1 above)
Comment (drivenflywheel) [reply src/main/java/emissary/server/api/Version.java]: Why would the response need those? (unless the response is being teed directly into a log file and we want those to be pat of the logged json payload)

**Comment by cfkoehler** [comment]
> Interesting side effect from the last version changes:
> 
> ```
> ./emissary version -q
> Emissary Version: 7.10.0-SNAPSHOT - built on 2022-10-25T12:17:47+0000 - git hash: 0fbdb52
> Emissary Version: 7.10.0-SNAPSHOT 2022-10-25T12:17:45Z
> ```

Resolved in: https://github.com/NationalSecurityAgency/emissary/pull/354

**Comment by cfkoehler** [comment]
TODO: 
- [ ] Validate that other components can override the version returned.
- [ ] Clean up tests and refactor

**Review by drivenflywheel — CHANGES_REQUESTED** [review]
Hmm.  I haven't tried to diff the current state of this code vs when it was first submitted, but here's the response I'm currently getting if I hit the versionDetails endpoint:

```bash
]$ curl -s -u emissary:emissary123 --anyauth http://localhost:8001/api/versionDetails | jq
{
  "errors": [],
  "response": {
    "Emissary": "emissary.client.response.MapResponseEntity@3056898e"
  }
}

```

**Comment by drivenflywheel** [comment]
> Hmm. I haven't tried to diff the current state of this code vs when it was first submitted, but here's the response I'm currently getting if I hit the versionDetails endpoint:
> 
> ```shell
> ]$ curl -s -u emissary:emissary123 --anyauth http://localhost:8001/api/versionDetails | jq
> {
>   "errors": [],
>   "response": {
>     "Emissary": "emissary.client.response.MapResponseEntity@3056898e"
>   }
> }
> ```

Similar result with the versionSimple endpoint

**Comment by cfkoehler** [comment]
Sorry should have put it back in draft. Started working on it again this morning and have not pushed up the latest changes yet. Experimenting on how to make it overridable in follow on applications.

**Pull request closed without merging**